// code derived from https://github.com/lodash/lodash/commit/d8e069cc3410082e44eb18fcf8e7f3d08ebe1d4a

// lodash license:
/**
 * @license
 * Lodash <https://lodash.com/>
 * Copyright JS Foundation and other contributors <https://js.foundation/>
 * Released under MIT license <https://lodash.com/license>
 * Based on Underscore.js 1.8.3 <http://underscorejs.org/LICENSE>
 * Copyright Jeremy Ashkenas, DocumentCloud and Investigative Reporters & Editors
 */

--- a/index.js
+++ b/index.js
@@ -589,6 +589,20 @@
   }
 
   /**
+   * Gets the value at `key`, unless `key` is "__proto__".
+   *
+   * @private
+   * @param {Object} object The object to query.
+   * @param {string} key The key of the property to get.
+   * @returns {*} Returns the property value.
+   */
+  function safeGet(object, key) {
+    return key == '__proto__'
+      ? undefined
+      : object[key];
+  }
+
+  /**
    * An implementation of `_.uniq` optimized for sorted arrays without support
    * for callback shorthands and `this` binding.
    *
@@ -2366,7 +2380,7 @@
           baseMergeDeep(object, source, key, baseMerge, customizer, stackA, stackB);
         }
         else {
-          var value = object[key],
+          var value = safeGet(object, key),
               result = customizer ? customizer(value, srcValue, key, object, source) : undefined,
               isCommon = result === undefined;
 
@@ -2399,7 +2413,7 @@
      */
     function baseMergeDeep(object, source, key, mergeFunc, customizer, stackA, stackB) {
       var length = stackA.length,
-          srcValue = source[key];
+          srcValue = safeGet(source, key);
 
       while (length--) {
         if (stackA[length] == srcValue) {
@@ -2407,7 +2421,7 @@
           return;
         }
       }
-      var value = object[key],
+      var value = safeGet(object, key),
           result = customizer ? customizer(value, srcValue, key, object, source) : undefined,
           isCommon = result === undefined;
 
